api/types/constants.go

```diff
@@ -1031,6 +1031,16 @@ const (
 	// that belong to different discovery services that operate on different sets of resources.
 	TeleportInternalDiscoveryGroupName = TeleportInternalLabelPrefix + "discovery-group-name"
 
+	// TeleportInternalDiscoveryIntegrationName is the label used to store the name of the integration
+	// whose credentials were used to discover the resource.
+	// It is used to report stats for a given Integration / DiscoveryConfig.
+	TeleportInternalDiscoveryIntegrationName = TeleportInternalLabelPrefix + "discovery-integration-name"
+
+	// TeleportInternalDiscoveryConfigName is the label used to store the name of the discovery config
+	// whose matchers originated the resource.
+	// It is used to report stats for a given Integration / DiscoveryConfig.
+	TeleportInternalDiscoveryConfigName = TeleportInternalLabelPrefix + "discovery-config-name"
```

Comment:
Isn't the dynamic discovery config independent of the integration?

Reply:
It is, but we only report stats for matchers using integrations.
So, that's why the comment says Integration / DiscoveryConfig.

Reply:
why not report DiscoveryConfig when there is no integration, if they are independant?

I feel the current impl is not very consistent. For example, a **dynamic** discovery config may have AWS matchers without the `integration`? Do we still set the `DiscoveryConfig` in that case? How about a **static** discovery config with an `integration` set in the AWS matcher?

Reply:
Maybe leave a TODO to report DiscoveryConfig for all fetchers?

Reply:
> why not report DiscoveryConfig when there is no integration, if they are independant?

You are right, I was narrowing too much on my use case, but I've now added the DiscoveryConfig to the Azure and GCP fetchers.

> How about a static discovery config with an integration set in the AWS matcher?

For those, we don't add the label.
We could, but it's currently out of scope because I'm focusing on building a dashboard based on a given integration.